internal/manager/pvcleanup/controller_test.go

```diff
@@ -0,0 +1,429 @@
+					Name:       "test-pv-2",
+					Finalizers: []string{ExternalProvisionerFinalizer},
+				},
+			},
+			finalizer: ExternalProvisionerFinalizer,
+			expected:  true,
+		},
+		{
+			name: "PV with both finalizers",
+			pv: &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pv-3",
+					Finalizers: []string{
+						PVProtectionFinalizer,
+						ExternalProvisionerFinalizer,
+					},
+				},
+			},
+			finalizer: PVProtectionFinalizer,
+			expected:  true,
+		},
+		{
+			name: "PV without finalizer",
+			pv: &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-pv-4",
+					Finalizers: []string{},
+				},
+			},
+			finalizer: PVProtectionFinalizer,
+			expected:  false,
+		},
+		{
+			name: "PV with different finalizer",
+			pv: &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "test-pv-5",
+					Finalizers: []string{"some-other-finalizer"},
+				},
+			},
+			finalizer: PVProtectionFinalizer,
+			expected:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := hasFinalizer(tt.pv, tt.finalizer)
+			if result != tt.expected {
+				t.Errorf("hasFinalizer() = %v, expected %v", result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestRemoveFinalizer(t *testing.T) {
+	tests := []struct {
+		name       string
+		finalizers []string
+		finalizer  string
+		expected   []string
+	}{
+		{
+			name:       "Remove single finalizer",
+			finalizers: []string{PVProtectionFinalizer},
+			finalizer:  PVProtectionFinalizer,
+			expected:   []string{},
+		},
+		{
+			name: "Remove one of multiple finalizers",
+			finalizers: []string{
+				PVProtectionFinalizer,
+				ExternalProvisionerFinalizer,
+			},
+			finalizer: PVProtectionFinalizer,
+			expected:  []string{ExternalProvisionerFinalizer},
+		},
+		{
+			name:       "Remove non-existent finalizer",
+			finalizers: []string{ExternalProvisionerFinalizer},
+			finalizer:  PVProtectionFinalizer,
+			expected:   []string{ExternalProvisionerFinalizer},
+		},
+		{
+			name:       "Remove from empty list",
+			finalizers: []string{},
+			finalizer:  PVProtectionFinalizer,
+			expected:   []string{},
+		},
+		{
+			name: "Remove from list with multiple same finalizers",
+			finalizers: []string{
+				PVProtectionFinalizer,
+				PVProtectionFinalizer,
+				ExternalProvisionerFinalizer,
+			},
+			finalizer: PVProtectionFinalizer,
+			expected:  []string{ExternalProvisionerFinalizer},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := removeFinalizer(tt.finalizers, tt.finalizer)
+			if len(result) != len(tt.expected) {
+				t.Errorf("removeFinalizer() returned %d finalizers, expected %d", len(result), len(tt.expected))
+				return
+			}
+			for i, finalizer := range result {
+				if finalizer != tt.expected[i] {
+					t.Errorf("removeFinalizer() finalizer[%d] = %s, expected %s", i, finalizer, tt.expected[i])
+				}
+			}
+		})
+	}
+}
```

Comment:
We can remove these unit tests now too, the one for removeFinalizer and hasFinalizer